src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -2383,16 +2393,34 @@ R"(#include <stdio.h>
             SET_INTRINSIC_NAME(Exp, "exp");
             SET_INTRINSIC_NAME(Exp2, "exp2");
             SET_INTRINSIC_NAME(Expm1, "expm1");
+            SET_INTRINSIC_NAME(SymbolicSymbol, "Symbol");
+            SET_INTRINSIC_NAME(SymbolicPi, "pi");
+            case (static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicAdd)): {
+                LCOMPILERS_ASSERT(x.n_args == 2);
+                this->visit_expr(*x.m_args[0]);
+                std::string arg1 = src;
+                this->visit_expr(*x.m_args[1]);
+                std::string arg2 = src;
+                out = arg1 + "," + arg2;
+                src = out;
+                break;
+            }
             default : {
                 throw LCompilersException("IntrinsicFunction: `"
                     + ASRUtils::get_intrinsic_name(x.m_intrinsic_id)
                     + "` is not implemented");
             }
         }
         headers.insert("math.h");
-        this->visit_expr(*x.m_args[0]);
-        out += "(" + src + ")";
-        src = out;
+        if (x.n_args == 0){
+            src = out;
+        } else if (x.n_args == 1) {
+            this->visit_expr(*x.m_args[0]);
+            if (x.m_intrinsic_id != static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicSymbol)) {
+                out += "(" + src + ")";
+                src = out;
+            }
+        }
```

Comment:
For Later: Instead of an `if-else` ladder, can we generalize this? For example:
```cpp
out = "(";
this->visit_expr(*x.m_args[0]);
out += src;
for (int i = 1; i < x.n_args; i++) {
    this->visit_expr(*x.m_args[i]);
    out += src;
}
out += ")";
```

Reply:
Yes , we would be moving towards something like this once we start introducing intrinsic function that deal with multiple arguments like more binary operators like `SymbolicPow` etc and algorithms ( like `limits` which would be intaking a `function` , `symbol` and a `point` about which the limit needs to be calculated as arguments)